### PR TITLE
Use setuptools-scm to generate version from SCM (fallback to static)

### PR DESCRIPTION
**Pull request by gaborbernat**
👍 Compared to what we have now, we're a bit ahead in the sense of post this we'll only have one version number within our repo, and the version number will be automatically generated from the SCM (if present). We'll need to make sure to update the version file before we do a release, but we're already here today.

Closes https://github.com/pypa/build/issues/170

### CODE REVIEW

**Review by FFY00 — CHANGES_REQUESTED** [review]
I have said I do not want to pull any other bootstraping dependencies.

**Comment by gaborbernat** [comment]
> I have said I do not want to pull any other bootstraping dependencies.

* ``bootstrapping``

Look man, this is not a hard dependency, it's a soft one. The package works just fine (as is today) without it. But if installed provide additional functionality today not present.

**Comment by pganssle** [comment]
Yeah, it has an additional dependency in the `pyproject.toml`, but anyone who cares about bootstrapping isn't blindly pulling in the `pyproject.toml` dependencies anyway.

For building from `sdist`, I think `setuptools_scm` isn't even really used, since there's no git for it to even interact with (though there's no way to indicate this in `pyproject.toml` without modifying the file in place as part of the sdist build).

**Comment by layday** [comment]
I assume the tests are failing because we need a `MANIFEST.in` for `version.fallback` for when the `setuptools_scm` file finder is not in use (i.e. without isolation). `setuptools_scm` will pull everything that's tracked by the SCM into the sdist automatically.  We might also want to customise `MANIFEST.in` to exclude files which wouldn't be included by vanilla `setuptools` if sdist parity is important.

**Review by FFY00 — APPROVED** [review]
It is still annoying for packagers, but sure.

Please properly document that `setuptools-scm` is not actually required for the build, and the project can be bootstrapped without it.

Other than that, do whatever.

**Comment by gaborbernat** [comment]
> It is still annoying for packagers, but sure.

Why? Can you expand on this a bit? 😊

**Comment by FFY00** [comment]
> Why? Can you expand on this a bit?

Not all packagers will know this, it takes a bit of digging to find out about this.

We should definitely document this.

File: src/build/__init__.py [hunk 1]
```diff
@@ -3,20 +3,20 @@
...
 import sys
 import warnings
-
 from typing import Iterator, Mapping, Optional, Sequence, Set, Text, Union
 
 import pep517.wrappers
 import toml
 import toml.decoder
 
+from .version import __version__
```
Comment (FFY00) [inline src/build/__init__.py:18]: Did you forget to commit `version.py`?

Comment (FFY00) [reply src/build/__init__.py:18]: nvm

**Review by webknjaz — APPROVED** [review]

**Comment by gaborbernat** [comment]
Will create new PR.

**Pull request closed without merging**